### DISCUSSION TITLE
getting full uri/url on getServerSideProps all parameters after an hash ("#") are being discarded

### DISCUSSION THREAD

**Discussion by albjeremias** [vercel/next.js#56967] [Help] [answered]
### Summary

Hi!

I'm using a window.replace for setting up parameters without having to refresh the page. I tried shallow routing didn't work.
'window.location.replace(`#?${urlParams.toString()}`);` 

so this creates an url like:
https://webpage/Explore#?parameter1=XXX

so the idea is that the user can easily copy and paste the url to someone, and that the  URL parameters to generate SEO tags.

when I use the code:
```
export const getServerSideProps = async (ctx: NextPageContext) => {
   console.log(ctx.params)
   console.log(ctx.url)
}
```
I don't get any parameters which are behind an hash.

I also tried to create a middleware:
```
import { NextResponse } from 'next/server'
import type { NextRequest } from 'next/server'
 
export function middleware(request: NextRequest) {
    // console.log(request.nextUrl.pathname)
    console.log(request.nextUrl.href)
}
```
didnt work.

any ideas?
[code](https://github.com/helpbuttons/helpbuttons/blob/main/web/src/pages/Explore/HoneyComb/index.tsx#L247)
[demo](https://dev.helpbuttons.org/Explore)

I'm developing https://dev.helpbuttons.org
check out the code in here: https://github.com/helpbuttons/helpbuttons

**Comment by FlyingTurkman** [comment]
I have been pull request about it
```jsx
import { useRouter } from 'next/navigation'

const router = useRouter()

useEffect(() => {
    if (!exploreSettings?.loading) {
      urlParams.append('zoom', exploreSettings.zoom);
      urlParams.append('lat', roundCoord(exploreSettings.center[0]));
      urlParams.append('lng', roundCoord(exploreSettings.center[1]));

      //window.location.replace(`#?${urlParams.toString()}`);
     router.replace(`#?${urlParams.toString()}`)
    }
  }, [exploreSettings]);
```

**Reply by albjeremias** [reply]
hi, thanks for the contribution.. but that is not solving the issue.. because the issue is not to put the values in the url.. the issue is to load the parameters when getServerSideProps is called. they get discarded from the URL

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Everything that comes after the `#` is never sent to the server. This is handled by your browser. The `#` is used as a client side only thing as per spec. 

More content: https://github.com/vercel/next.js/issues/55044#issuecomment-1708001099

**Reply by albjeremias** [reply]
that's what i thought . any ideas on how-to work around this?

is it possible.. to have a url changing.. and then use it when copy/paste with getServerSideProps?

**Reply by icyJoseph** [reply]
> is it possible.. to have a url changing.. and then use it when copy/paste with getServerSideProps?

Could you give a bit more **developed** use case?

**Reply by albjeremias** [reply]
here's a full example:
https://dev.helpbuttons.org/Explore

when you go to the Explore.. and you are navigating.. the URL keeps changing with the coordinates.. and now im developing a new feature when you open a "button" (on the left side list), it will be added to the URL. 

so the user when copies the URL.. and pastes.. it should generate SEO with getServerSideProps.. but the button id is not detected on the server.. because its store in the URI after "#".. you understand better?

**Reply by icyJoseph** [reply]
Shouldn't it just be ok to use `queries`?  without `#`?

**Reply by albjeremias** [reply]
when using queries.. the page is redirected.. and when going back to "Explore" the scroll position of the list is lost
